jump-core/src/main/java/com/bitdecay/jump/BitWorld.java

```diff
@@ -179,6 +236,57 @@ private void resolveAndApplyPendingResolutions() {
 		pendingResolutions.clear();
 	}
 
+	private void expireContact(BitBody body) {
+		Iterator<BitBody> iterator = contacts.get(body).iterator();
+		BitBody otherBody = null;
+		while(iterator.hasNext()) {
+			otherBody = iterator.next();
+			if (GeomUtils.intersection(body.aabb, otherBody.aabb) == null) {
+				iterator.remove();
+				for (ContactListener listener : body.getContactListeners()) {
+					listener.contactEnded(otherBody);
+				}
+				for (ContactListener listener : otherBody.getContactListeners()) {
+					listener.contactEnded(body);
+				}
+			}
+		}
+	}
+
+	private void findNewContact(BitBody body) {
+		// TODO: collide vs Static and Dynamic dynamicBodies and report them via a callback (preferably)
+		// We need to update each body against the level grid so we only collide things worth colliding
+		BitPoint startCell = body.aabb.xy.floorDivideBy(tileSize, tileSize).minus(gridOffset);
+
+		int endX = (int) (startCell.x + Math.ceil(1.0 * body.aabb.width / tileSize));
+		int endY = (int) (startCell.y + Math.ceil(1.0 * body.aabb.height / tileSize));
+
+		for (int x = (int) startCell.x; x <= endX; x++) {
+			if (!occupiedSpaces.containsKey(x)) {
+				continue;
+			}
+			for (int y = (int) startCell.y; y <= endY; y++) {
+				if (!occupiedSpaces.get(x).containsKey(y)) {
+					continue;
+				}
+				for (BitBody otherBody : occupiedSpaces.get(x).get(y)) {
```

Comment:
Is this something that can be streamed? Or maybe streams aren't the way to go with a system that cares about performance?